mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java

```diff
@@ -338,10 +341,29 @@ private void processIntent(Intent intent) {
             } else if (intent.getAction().equals("android.intent.action.VIEW")) {
                 Log.d(TAG, "This is URL Action");
                 mNfcData = intent.getDataString();
+            } else if (intent.getAction().equals(CUSTOM_BROADCAST_RECEIVER_INTENT)) {
+                Log.d(TAG, "Boot broadcast");
+                setupCustomBroadcastReceiver();
             }
         }
     }
 
+    private void setupCustomBroadcastReceiver() {
+        // Custom broadcast receiver
+        if (mSettings.getBoolean(Constants.PREFERENCE_CUSTOM_BROADCAST, false)) {
+            Log.d(TAG, "start cbr");
+            Intent startIntent = new Intent(this, ForegroundService.class);
+            startIntent.setAction(Constants.ACTION.STARTFOREGROUND_ACTION);
+            startIntent.putExtra("broadcast", mSettings.getString(Constants.PREFERENCE_CUSTOM_BROADCAST_BROADCAST, ""));
```

Comment:
Can't the service figure out the broadcast it should listen to by itself?

Reply:
My idea was to pass all information (broadcast, item to change, intent extra, openhab url) from mainactivity to foreground service and from there to CustomBroadcastListenerService. But I dont have an idea how I can get those infos in CustomBroadcastReceiver#onReceive()

Reply:
CustomBroadcastListenerService now reads all settings and writes them to CustomBroadcastReceiver.